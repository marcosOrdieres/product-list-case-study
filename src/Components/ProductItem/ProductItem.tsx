import React, { useState } from 'react';
import styled from 'styled-components';
import { ProductType } from '../../interfaces/Product.interface';
import MainImageContainer from '../MainContainerImage/MainContainerImage';
import Image from '../Image/Image';

export interface ColumnProps {
  noVerticalLine?: boolean,
  bottomLine?: boolean,
};

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  width:1500px;
  cursor:pointer;
`;

const Column = styled.div<ColumnProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  flex:1;
  border-left:${(props) => props.noVerticalLine ? null : '1px solid rgba(12, 15, 20, 0.3)'};
  border-bottom:${(props) => !props.bottomLine ? null : '1px solid rgba(12, 15, 20, 0.3)'};
`;

const Row = styled.div`
  margin-right: 10px;
`;

const Center = styled.div`
  text-align: center;
  align-items: center;
  justify-content: center;
`;

export interface ProductItemProps {
  product?: ProductType,
};

const ProductItem = ({ product }: any) => {

  const [additionalPhotosOpen, setAdditionalPhotosOpen] = useState<boolean>(false)
  return (
    <>
      <Container onClick={() => setAdditionalPhotosOpen(!additionalPhotosOpen)}>

        <Column noVerticalLine bottomLine={additionalPhotosOpen}>
          <Row>{product.title}</Row>
        </Column>

        <Column bottomLine={additionalPhotosOpen}>
          <Row>{product.gtin}</Row>
        </Column>

        <Column bottomLine={additionalPhotosOpen}>
          <Row>{product.gender}</Row>
        </Column>

        <Column bottomLine={additionalPhotosOpen}>
          <Row>{product.price}</Row>
        </Column>
        <Column bottomLine={additionalPhotosOpen}>
          <Row>{product.sale_price}</Row>
        </Column>

        <Column bottomLine={additionalPhotosOpen}>
          <MainImageContainer>
            <Image src={product.image_link} />
          </MainImageContainer>
        </Column>
      </Container>

      {additionalPhotosOpen && (
        <>
          <Center>
            <p>Additional Images</p>
          </Center>

          <Container>
            <Column noVerticalLine>
              {product.additional_image_link.split(',').map((value: string) => (
                <MainImageContainer padding>
                  <Image src={value} />
                </MainImageContainer>
              ))}
            </Column>
          </Container>
        </>
      )}
    </>
  )
};

export default ProductItem;