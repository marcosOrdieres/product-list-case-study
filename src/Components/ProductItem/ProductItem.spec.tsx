import { render } from '@testing-library/react';
import React from 'react';

import ProductItem from './ProductItem';

const product = {
  additional_image_link: ['example1.com', 'example2.com', 'example3.com'],
  gender: 'binary',
  gtin: '1',
  image_link: 'example.com',
  price: '20',
  sale_price: '20',
  title: 'title'
}

describe('Product Items', () => {
  it('should match styles', () => {
    const { asFragment } = render(
      <ProductItem product={product} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
