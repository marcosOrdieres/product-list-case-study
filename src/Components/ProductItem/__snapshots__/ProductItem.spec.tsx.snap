// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Items should match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnuX gjKEHZ"
  >
    <div
      class="sc-hKgJUU fQFrfy"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        title
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        1
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        binary
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-bdfBQB kjmAOv"
      >
        <img
          class="sc-gsTEea kWaMru"
          src="example.com"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Product Items with the case of additionalPhotosOpen = false should match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnuX gjKEHZ"
  >
    <div
      class="sc-hKgJUU fQFrfy"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        title
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        1
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        binary
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-bdfBQB kjmAOv"
      >
        <img
          class="sc-gsTEea kWaMru"
          src="example.com"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Product Items with the case of additionalPhotosOpen = true should match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnuX gjKEHZ"
  >
    <div
      class="sc-hKgJUU fQFrfy"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        title
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        1
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        binary
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-eCstlR hncgct"
      >
        20
      </div>
    </div>
    <div
      class="sc-hKgJUU cpXOpV"
    >
      <div
        class="sc-bdfBQB kjmAOv"
      >
        <img
          class="sc-gsTEea kWaMru"
          src="example.com"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
